source/extensions/common/wasm/context.cc

```diff
@@ -1407,7 +1407,8 @@ void Context::onGrpcReceiveTrailingMetadata(uint32_t token, Http::HeaderMapPtr&&
 }
 
 WasmResult Context::defineMetric(MetricType type, absl::string_view name, uint32_t* metric_id_ptr) {
-  auto stat_name = wasm_->stat_name_set_->add(name);
+  Stats::StatNameManagedStorage storage(name, wasm()->scope_->symbolTable());
+  auto stat_name = storage.statName();
```

Comment:
s/auto/Stats::StatName/

https://google.github.io/styleguide/cppguide.html#Type_deduction

the type of this is obvious to me personally but will not necessarily be to readers of this code, as it is not derived from context. The type is not onerous either so I think the guideline clearly states that explicit type is preferred here.

Reply:
Done: nit and added a TODO.